lib/ash_archival/resource/archive_related_arguments.ex

```diff
@@ -0,0 +1,10 @@
+defmodule AshArchival.ArchiveRelatedArguments do
+  @moduledoc """
+  The behaviour for specifiying arguments for related resources
+  """
+  @callback arguments(
+              original_arguments :: map(),
```

Comment:
I think it should take the original changeset, not just its arguments.

Reply:
We use only the arguments because it's not clear what would happen on a bulk_destroy otherwise.